docs/test_marking.md

```diff
@@ -0,0 +1,183 @@
+```python
+    @pytest.mark.component
+    @pytest.mark.unit
+```
+
+<a id="Priority"></a>
+#### Priority 
+Test priorities: low, medium, high, e.g.,
+```python
+     @pytest.mark.priority_low
+     @pytest.mark.priority_medium
+     @pytest.mark.priority_high
+```
+<a id="Component"></a>
+#### Component
+
+Component marking used for indication of different system components e.g.
+```python
+    @pytest.mark.components(DatumaroComponent.Datumaro)
+```
+<a id="Skip"></a>
+#### Skip 
+
+For marking tests, which should be skipped for example for not yet tests, e.g.,
+```python
+   @pytest.mark.skip(reason=SkipMessages.NOT_IMPLEMENTED)
+```
+<a id="Bug"></a>
+#### Bug(s):
+
+In case of test failure, bug should be entered into github issues, and test can be marked e.g.
+```python
+    @pytest.mark.bugs("DATUM-219 - Return format is not uniform")
+```
+<a id="Parameters"></a>
+#### Parameters: 
+
+Parameters are used for running the same test with different parameters e.g. 
+```python
+@pytest.mark.parametrize("numpy_array, batch_size", [  
+        (np.zeros([2]), 0),  
+        (np.zeros([2]), 1),
+        (np.zeros([2]), 2),
+        (np.zeros([2]), 5),
+        (np.zeros([5]), 2),
+    ])
+```
+(be aware that test parametrization is supported in pytest, and is not supported in unittests) 
+
+<a id="Fixtures"></a>
+#### Fixtures 
+
+If needed pytest fixtures can be used - for more details see pytest documentation <br>
+https://docs.pytest.org/en/6.2.x/contents.html <br>
+(be aware that fixtures are supported in pytest, and they are not supported in unittests) 
+
+<a id="TestNaming"></a>
+### Test naming convention
+
+<a id="TestName"></a>
+#### Test name:
+
+Test method should be prefixed with "test_" prefix e.g.   
+```python
+    test_*()
+```
+"test_" prefix is an indication for pytest to treat method as a test for execution. 
+It's important not to start other methods with the "test_" prefix.
+
+<a id="DestDoc"></a>
+### Test documentation
+
+<a id="Docstring"></a>
+#### Docstring 
+
+Tests are documented with Docstring. Every test method documentation string should contain: Description, Expected results 
+and Steps. These fields are required but, not limited e.g.
+```python
+        def test_can_convert_polygons_to_mask(self):
+        """
+        <b>Description:</b>
+        Ensure that the dataset polygon segmentation mode can be properly converted into dataset mask segmentation mode.
+
+        <b>Expected results:</b>
+        Dataset segmentation mask converted from dataset segmentation polygon is equal to expected mask.
+
+        <b>Steps:</b>
+        1. Prepare dataset with polygon segmentation mode (source dataset)
+        2. Prepare dataset with expected mask segmentation mode (target dataset)
+        3. Convert source dataset to target, with segmentation mode changed from polygon to mask and verify that result
+        segmentation mask is equal to expected mask.
```

Comment:
I feel it can quickly become a story similar to the story with comments in code. If a test is so complex it is impossible to understand quickly, it must be split and simplified. For many unit tests such description will be equal, or even bigger that the test code itself. It might be ok for complex or long integration/e2e tests, which are supposed to be like this.

Reply:
This high level description is dedicated to reports viewers. Viewer don't need to know particular programming language to have understanding what was tested and how.